tests/e2e/test_splunk_addon.py

```diff
@@ -737,5 +737,50 @@ def empty_method():
         skipped=len(constants.TA_REQ_TRANSITION_SKIPPED),
     )
 
-    # make sure that that we get a non '0' exit code for the testsuite as it contains failure
+    # make sure that we get a non '0' exit code for the testsuite as it contains failure
     assert result.ret == 0, "result not equal to 0"
+
+
+@pytest.mark.test_infinite_loop_fixture
+@pytest.mark.external
+def test_infinite_loop_in_ingest_data_fixture(testdir, request):
+    """Make sure that pytest accepts our fixture."""
+
+    testdir.makepyfile(
+        """
+        from pytest_splunk_addon.standard_lib.addon_basic import Basic
+        class Test_App(Basic):
+            def empty_method():
+                pass
+    """
+    )
+
+    shutil.copytree(
+        os.path.join(testdir.request.fspath.dirname, "addons/TA_fiction_indextime"),
+        os.path.join(testdir.tmpdir, "package"),
+    )
+
+    shutil.copytree(
+        os.path.join(testdir.request.fspath.dirname, "test_data_models"),
+        os.path.join(testdir.tmpdir, "tests/data_models"),
+    )
+
+    setup_test_dir(testdir)
+    SampleGenerator.clean_samples()
+    Rule.clean_rules()
+
+    # run pytest with the following cmd args
+    # we are providing wrong sc4s service details here so that we can recreate scenario where first worked raises execption and other workers get stuck
+    result = testdir.runpytest(
+        "--splunk-app=addons/TA_fiction_indextime",
+        "--splunk-type=external",
+        "--splunk-host=splunk",
+        "--splunk-data-generator=tests/addons/TA_fiction_indextime/default",
+        "--sc4s-host=splunk",
+        "--sc4s-port=100",
+        "-n 2",
+    )
+
+    # Here we are not interested in the failures or errors,
+    # we are basically checking that we get results and test execution does not get stuck
+    assert result.parseoutcomes().get("passed") > 0
```

Comment:
I don't think so, because that is handled internally via xdist. But everything would run on either of the workers.

Reply:
What about checking the logs if those are available somewhere?

My point is - if you don't make a fix - will this test pass anyways?

Reply:
Okay, So I actually tested it on main branch without making the fix and test kept on running.
Let me make the same in github actions, I will provide a link here for the actions run which should go on for 6hrs.

Here is the link with test change in main branch without fix: https://github.com/splunk/pytest-splunk-addon/actions/runs/8753677154/job/24023897918

This will go on forever till runner timeout.